src/components/GameBoard.js

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useEffect } from 'react';
+import { Card } from './Card'
+import '../styles/gameBoard.scss'
+import { v4 as uuidv4 } from 'uuid';
+
+const cardPool = [
+    {
+        type: 'faceGrin',
+        image: '😃'
+    },
+    {
+        type: 'faceSquint',
+        image: '😆'
+    },
+    {
+        type: 'animalOrangutan',
+        image: '🦧'
+    },
+    {
+        type: 'faceHalo',
+        image: '😇'
+    },
+    {
+        type: 'faceWink',
+        image: '😉'
+    },
+    {
+        type: 'animalGorilla',
+        image: '🦍'
+    },
+    {
+        type: 'animalPoodle',
+        image: '🐩'
+    }
+    ,
+    {
+        type: 'animalMouse',
+        image: '🐭'
+    }
+];
+
+const createShuffledDeck = deck => {
+    deck = [...deck, ...deck];
+    let count = deck.length;
+    while (count > 0) {
+        let randomPick = Math.floor(Math.random() * count);
+        count--;
+        [deck[count], deck[randomPick]] = [deck[randomPick], deck[count]];
+    }
+    return deck.map(each => ({ ...each, 'id': uuidv4() }));
+}
+
+export const GameBoard = () => {
+    const [inactiveCards, setInactiveCards] = useState([]);
+    const [disableAllClick, setDisableAllClick] = useState(false);
+    const [openCards, setOpenCards] = useState([]);
+    const [numberOfMoves, setNumberOfMoves] = useState(0);
+    const [cards, setCards] = useState(() =>
+        createShuffledDeck(cardPool)
+    );
+
+    useEffect(() => {
+        if (openCards.length === 2) {
+            setDisableAllClick(true);
+            setTimeout(() => { evaluate() }, 3500);
+            setDisableAllClick(true);
+        }
+    }, [openCards])
+
+    const restartGame = () => {
+        setCards(createShuffledDeck(cardPool));
+        setInactiveCards([]);
+        setOpenCards([]);
+        setNumberOfMoves(0);
+    }
+
+    const handleCardClicked = (clickedCardId) => {
```

Comment:
But if I store the complete card object on each `handleCardClicked`, hence storing the whole card object into `openedCards` and `inactiveCards` instead of storing only their IDs, then in the current html `onClick={() => handleCardClicked(card.id)}` and` isFlipped={openedCards.includes(card.id)}` have to become something like `openedCards.find(eachCardObjectInDeck => eachCardObjectInDeck.id === card.id )` for both openedCards and inactiveCards.

Or I can change the `checkCardInfo` function to be just a filter function, and let `evaluate` function compares their types: 

```
  const checkCardInfo = (cardId) => {
    let cardObject = cards.filter(card => card.id === cardId);
    return cardObject;
  }

  const evaluate = () => {
    const [card1, card2] = openedCards;
    if ((checkCardInfo(card1))[0].type === (checkCardInfo(card2))[0].type) {
      setInactiveCards([...inactiveCards, card1, card2])
    }
    setDisableAllClick(false);
    setOpenedCards([])
  }
```

This might be less code overall, compare to storing the whole object, and have to loop through the object all the time.

Reply:
You make a good point. Storing id is probably a better choice here.